src/bun.js/webcore/response.zig

```diff
@@ -1850,7 +1850,12 @@ pub const Fetch = struct {
         var unix_socket_path: ZigString.Slice = ZigString.Slice.empty;
 
         var url_proxy_buffer: []const u8 = "";
-        var is_file_url = false;
+        const URLType = enum {
```

Comment:
maybe we could just add .type() in `ZigURL` since we use `url.isFile()`, `url.isBlob()` etc, will be less error prone.